db/migrate/20231019215755_create_languages.rb

```diff
@@ -0,0 +1,9 @@
+class CreateLanguages < ActiveRecord::Migration[7.0]
+  def change
+    create_table :languages, id: :binary, limit: 16, force: :cascade do |t|
+      t.string :name, null: false
```

Comment:
https://github.com/rmm5t/iso639 : https://github.com/xwmx/iso-639
_both under-maintained from a quick look_

Source data files.
https://github.com/rmm5t/iso639/blob/master/lib/iso639/ISO-639-2_utf-8.txt
https://github.com/xwmx/iso-639/blob/master/lib/data/ISO-639-2_utf-8.txt

I think we should include (and populate in a `db:seed` via `db/seeds.rb`) a handful of common languages.

```suggestion
      t.string :english_name, null: false
      t.string :french_name, null: false
      t.string :local_name, null: false
      t.string :iso639_alpha2, null: false
      t.string :iso639_alpha3, null: false # maybe make this the primary key if that's easy?
```

There's also:

https://github.com/grosser/i18n_data

for local names in future.

Reply:
I ended up just copy pasting the source file in lib/data/iso_languages.txt and a script in db/seeds.rb to populate common languages.

Copied over https://github.com/xwmx/iso-639/blob/master/lib/data/ISO-639-2_utf-8.txt because it had 1 more line.

Lots of googling and talking to GPT to get my head around this but seems this approach is reasonable.

--
RE: primary key:

Didn't make :iso639_alpha3 primary key because we were using ULIDs in other tables. 

So ins user_languages it was easy to do validate uniqueness by language_id. Is there a reason/benefit to making this the primary key? And having different data types as primary keys in different tables won't causes issues down the line?

Reply:
Results after Running rails db:seed

<img width="2045" alt="image" src="https://github.com/goodscary/firstrubyfriend/assets/29284404/74ca2898-1713-4fb5-936e-d503cccf93bb">